### PR TITLE
[#9595] Add tests for POST /session/publish

### PR DESCRIPTION
**Pull request by erjillsison**
Part of #9595

Create tests in `PublishFeedbackSessionActionTest.java`

### CODE REVIEW

**Comment by erjillsison** [comment]
https://github.com/TEAMMATES/teammates/blob/5eff2f4d0819f8910f95f8da04ca2d60b1eb0bb4/src/main/java/teammates/ui/webapi/action/PublishFeedbackSessionAction.java#L39-L50

The `EntityDoesNotExistException` seems to never be thrown because `logic.publishFeedbackSession(session);` in line 42 will throw an `AssertionError` instead if session is null.

**Comment by xpdavid** [comment]
Yup. You are right. I think the method signature of `publishFeedbackSession` is problematic. It should accept courseId and sessionName instead of a concrete feedbackSessionAttribute. Feel free to fire an issue and fix it.

**Comment by erjillsison** [comment]
Will put this on hold until issue #9734 is fixed

**Comment by xpdavid** [comment]
@erjillsison #9734 is fixed and merged.

File: src/test/java/teammates/test/cases/webapi/PublishFeedbackSessionActionTest.java [hunk 1]
```diff
@@ -23,13 +30,90 @@ protected String getRequestMethod() {
...
+    }
+
+    @Test
+    public void testExecute_invalidRequests_shouldFail() {
+        ______TS("non existent session name");
+
+        CourseAttributes course = typicalBundle.courses.get("typicalCourse1");
+        FeedbackSessionAttributes session = typicalBundle.feedbackSessions.get("session1InCourse1");
+        String randomSessionName = "randomName";
+        assertNotNull(course.getId(), session.getFeedbackSessionName());
```
Comment (xpdavid) [inline src/test/java/teammates/test/cases/webapi/PublishFeedbackSessionActionTest.java]: What does the assertion try to do here?

BTW, the first parameter of assertNotNull is `message` not `object`. There is no such assertion method for asserting two non-null objects.

File: src/test/java/teammates/test/cases/webapi/PublishFeedbackSessionActionTest.java [hunk 2]
```diff
@@ -23,13 +30,90 @@ protected String getRequestMethod() {
...
+
+        assertNull(logic.getFeedbackSession(randomSessionName, course.getId()));
+
+        PublishFeedbackSessionAction publishFeedbackSessionAction = getAction(params);
+        JsonResult result = getJsonResult(publishFeedbackSessionAction);
+        MessageOutput output = (MessageOutput) result.getOutput();
+
+        assertEquals(String.format("Trying to update a non-existent feedback session: %s/%s",
+                course.getId(), randomSessionName), output.getMessage());
+        assertEquals(result.getStatusCode(), HttpStatus.SC_INTERNAL_SERVER_ERROR);
```
Comment (xpdavid) [inline src/test/java/teammates/test/cases/webapi/PublishFeedbackSessionActionTest.java]: Actually, we should return 404. Can be corrected in another issue.

File: src/test/java/teammates/test/cases/webapi/PublishFeedbackSessionActionTest.java [hunk 3]
```diff
@@ -23,13 +30,90 @@ protected String getRequestMethod() {
...
+                Const.ParamsNames.FEEDBACK_SESSION_NAME, session.getFeedbackSessionName(),
+        };
+
+        PublishFeedbackSessionAction publishFeedbackSessionAction = getAction(params);
+
+        JsonResult result = getJsonResult(publishFeedbackSessionAction);
+        FeedbackSessionData feedbackSessionData = (FeedbackSessionData) result.getOutput();
+
+        assertEquals(feedbackSessionData.getFeedbackSessionName(), session.getFeedbackSessionName());
+        assertEquals(feedbackSessionData.getPublishStatus(), FeedbackSessionPublishStatus.PUBLISHED);
```
Comment (xpdavid) [inline src/test/java/teammates/test/cases/webapi/PublishFeedbackSessionActionTest.java]: The data return by the sever is verified. However, you should add another assertion to verify the status of the feedback session in DB. (e.g. use `logic.getFeedbackSession()` and verify the publish status).

Comment (erjillsison) [reply src/test/java/teammates/test/cases/webapi/PublishFeedbackSessionActionTest.java]: Updated

File: src/test/java/teammates/test/cases/webapi/PublishFeedbackSessionActionTest.java (hunk 2 above)
Comment (erjillsison) [reply src/test/java/teammates/test/cases/webapi/PublishFeedbackSessionActionTest.java]: Roger. I'll create the issue

File: src/test/java/teammates/test/cases/webapi/PublishFeedbackSessionActionTest.java [hunk 4]
```diff
@@ -23,13 +30,92 @@ protected String getRequestMethod() {
...
+                Const.ParamsNames.COURSE_ID, course.getId(),
+                Const.ParamsNames.FEEDBACK_SESSION_NAME, session.getFeedbackSessionName(),
+        };
+
+        PublishFeedbackSessionAction publishFeedbackSessionAction = getAction(params);
+
+        JsonResult result = getJsonResult(publishFeedbackSessionAction);
+        FeedbackSessionData feedbackSessionData = (FeedbackSessionData) result.getOutput();
+
+        assertEquals(feedbackSessionData.getFeedbackSessionName(), session.getFeedbackSessionName());
```
Comment (xpdavid) [inline src/test/java/teammates/test/cases/webapi/PublishFeedbackSessionActionTest.java]: The first parameter if assertEquals should be expected value not actual value. Check other places and other PR also :P

Comment (erjillsison) [reply src/test/java/teammates/test/cases/webapi/PublishFeedbackSessionActionTest.java]: Updated

**Pull request merged**